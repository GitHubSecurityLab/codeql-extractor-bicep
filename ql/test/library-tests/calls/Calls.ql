private import bicep

query predicate call(CallExpression callable) { any() }

query predicate callArguments(CallExpression call, Expr argument) { call.getArguments() = argument }
