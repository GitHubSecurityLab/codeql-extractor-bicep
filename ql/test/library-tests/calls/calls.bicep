param siteName string = 'site${uniqueString(resourceGroup().id)}'
param hostingPlanName string = '${siteName}-plan'

func buildUrl(https bool, hostname string, path string) string => '${https ? 'https' : 'http'}://${hostname}${empty(path) ? '' : '/${path}'}'

func sayHelloString(name string) string => 'Hi ${name}!'

func sayHelloObject(name string) object => {
  hello: 'Hi ${name}!'
}

func nameArray(name string) array => [
  name
]

func addNameArray(name string) array => [
  'Mary'
  'Bob'
  name
]

output azureUrl string = buildUrl(true, 'microsoft.com', 'azure')
output greetingArray array = map(['Evie', 'Casper'], name => sayHelloString(name))
output greetingObject object = sayHelloObject('John')
output nameArray array = nameArray('John')
output addNameArray array = addNameArray('John')
