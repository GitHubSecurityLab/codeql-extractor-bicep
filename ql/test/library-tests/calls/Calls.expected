call
| calls.bicep:1:32:1:63 | CallExpression |
| calls.bicep:1:45:1:59 | CallExpression |
| calls.bicep:4:111:4:121 | CallExpression |
| calls.bicep:22:26:22:65 | CallExpression |
| calls.bicep:23:30:23:82 | CallExpression |
| calls.bicep:23:62:23:81 | CallExpression |
| calls.bicep:24:32:24:53 | CallExpression |
| calls.bicep:25:26:25:42 | CallExpression |
| calls.bicep:26:29:26:48 | CallExpression |
callArguments
| calls.bicep:1:32:1:63 | CallExpression | calls.bicep:1:45:1:62 | MemberExpression |
| calls.bicep:1:32:1:63 | CallExpression | calls.bicep:1:45:1:62 | MemberExpression |
| calls.bicep:1:32:1:63 | CallExpression | calls.bicep:1:45:1:62 | MemberExpression |
| calls.bicep:4:111:4:121 | CallExpression | calls.bicep:4:117:4:120 | path |
| calls.bicep:4:111:4:121 | CallExpression | calls.bicep:4:117:4:120 | path |
| calls.bicep:4:111:4:121 | CallExpression | calls.bicep:4:117:4:120 | path |
| calls.bicep:22:26:22:65 | CallExpression | calls.bicep:22:35:22:38 | true |
| calls.bicep:22:26:22:65 | CallExpression | calls.bicep:22:35:22:38 | true |
| calls.bicep:22:26:22:65 | CallExpression | calls.bicep:22:35:22:38 | true |
| calls.bicep:22:26:22:65 | CallExpression | calls.bicep:22:41:22:55 | String |
| calls.bicep:22:26:22:65 | CallExpression | calls.bicep:22:41:22:55 | String |
| calls.bicep:22:26:22:65 | CallExpression | calls.bicep:22:41:22:55 | String |
| calls.bicep:22:26:22:65 | CallExpression | calls.bicep:22:58:22:64 | String |
| calls.bicep:22:26:22:65 | CallExpression | calls.bicep:22:58:22:64 | String |
| calls.bicep:22:26:22:65 | CallExpression | calls.bicep:22:58:22:64 | String |
| calls.bicep:23:30:23:82 | CallExpression | calls.bicep:23:34:23:51 | Array |
| calls.bicep:23:30:23:82 | CallExpression | calls.bicep:23:34:23:51 | Array |
| calls.bicep:23:30:23:82 | CallExpression | calls.bicep:23:34:23:51 | Array |
| calls.bicep:23:30:23:82 | CallExpression | calls.bicep:23:54:23:81 | LambdaExpression |
| calls.bicep:23:30:23:82 | CallExpression | calls.bicep:23:54:23:81 | LambdaExpression |
| calls.bicep:23:62:23:81 | CallExpression | calls.bicep:23:77:23:80 | name |
| calls.bicep:23:62:23:81 | CallExpression | calls.bicep:23:77:23:80 | name |
| calls.bicep:23:62:23:81 | CallExpression | calls.bicep:23:77:23:80 | name |
| calls.bicep:24:32:24:53 | CallExpression | calls.bicep:24:47:24:52 | String |
| calls.bicep:24:32:24:53 | CallExpression | calls.bicep:24:47:24:52 | String |
| calls.bicep:24:32:24:53 | CallExpression | calls.bicep:24:47:24:52 | String |
| calls.bicep:25:26:25:42 | CallExpression | calls.bicep:25:36:25:41 | String |
| calls.bicep:25:26:25:42 | CallExpression | calls.bicep:25:36:25:41 | String |
| calls.bicep:25:26:25:42 | CallExpression | calls.bicep:25:36:25:41 | String |
| calls.bicep:26:29:26:48 | CallExpression | calls.bicep:26:42:26:47 | String |
| calls.bicep:26:29:26:48 | CallExpression | calls.bicep:26:42:26:47 | String |
| calls.bicep:26:29:26:48 | CallExpression | calls.bicep:26:42:26:47 | String |
